cylc/flow/simulation.py

```diff
@@ -250,16 +195,22 @@ def sim_time_check(
 
     Returns:
         True if _any_ simulated task state has changed.
-    """
 
+    """
     sim_task_state_changed = False
     now = time()
     for itask in itasks:
         if broadcast_mgr:
             broadcast = broadcast_mgr.get_broadcast(itask.tokens)
             if broadcast:
-                update_nested_dict(
-                    itask.tdef.rtconfig, broadcast)
+                for config in SIMULATION_CONFIGS:
+                    if (
+                        config in broadcast
+                        and isinstance(broadcast[config], dict)
+                    ):
+                        itask.tdef.rtconfig[config].update(broadcast[config])
```

Comment:
This looks like it's modifying the task defs runtime config object, this would persist broadcasts after they've expired or been cancelled.

Reply:
Yes. Looks like I might have some re-thinking to do.

Reply:
Maybe take a look at how it's done for tasks run in live mode and see if we can use the same approach for simulation/dummy mode.